### PR TITLE
#5153 add UI tests for complex devTool operations

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #5153)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/cypress/support/customCommands.ts [hunk 1]
```diff
@@ -267,15 +267,25 @@ Cypress.Commands.add(
   },
 );
 
-Cypress.Commands.add('deleteAllServices', async () => {
-  await deleteAllServices();
+Cypress.Commands.add('deleteAllServices', () => {
+  cy.log(`Begin delete all services...`)
+    .then(async () => await deleteAllServices())
+    .then(() => cy.log(`Finish delete all services!`));
```
Comment (saivirtue) [inline ohara-manager/client/cypress/support/customCommands.ts:273]: wrap the native API calls into cypress command and make sure we could rely on the order of cypress commands

File: ohara-manager/client/cypress/support/customCommands.ts [hunk 2]
```diff
@@ -267,15 +267,25 @@ Cypress.Commands.add(
...
+    // check whether we are in the homepage or not
+    if ($body.find('#navigator').length === 0) {
+      // force to visit the root path
+      cy.visit('/');
+    }
+    cy.get('#navigator').within(() => {
+      cy.get('button').should('have.length', 1).click();
+    });
   });
+
```
Comment (saivirtue) [inline ohara-manager/client/cypress/support/customCommands.ts:288]: make sure the entry always from homepage :)

File: ohara-manager/client/src/components/DevTool/Body/View/ViewTopic.js [hunk 3]
```diff
@@ -76,7 +76,7 @@ const ViewTopic = () => {
                       ? isEmpty(message.value[header])
                         ? null
                         : JSON.stringify(message.value[header])
-                      : message.value[header]}
+                      : toString(message.value[header])}
```
Comment (saivirtue) [inline ohara-manager/client/src/components/DevTool/Body/View/ViewTopic.js:79]: the value of cell could be `boolean` and `<Cell>` could not render these values normally for us

File: ohara-manager/client/src/components/DevTool/Header/Controller/ControllerLog.js [hunk 4]
```diff
@@ -211,6 +215,7 @@ const ControllerLog = () => {
             <StyledTextField
               defaultValue={endTime || moment().format('YYYY-MM-DD[T]hh:mm')}
               disabled
+              id="End date"
```
Comment (konekoya) [inline ohara-manager/client/src/components/DevTool/Header/Controller/ControllerLog.js:218]: Normally, we use a "dash" between IDs: `end-date` :)

File: ohara-manager/client/cypress/support/pipelineCommands.ts [hunk 5]
```diff
@@ -183,8 +183,10 @@ Cypress.Commands.add('addElement', (name, kind, className) => {
       cy.findAllByText(/^add$/i).filter(':visible').click();
     }
 
-    // wait a little time for the cell added
-    cy.wait(3000);
+    // wait for the cell added
+    cy.get('#outline').within(() => {
+      cy.findByText(name).should('exist');
+    });
```
Comment (konekoya) [inline ohara-manager/client/cypress/support/pipelineCommands.ts:189]: Good call 👍

File: ohara-manager/client/cypress/it/appBar/devTool.test.ts [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+                'equal',
+                `Customize from ${$statEl
+                  .val()
+                  ?.toString()
+                  .replace('T', ' ')
+                  .replace(/-/g, '/')} to ${$endEl
+                  .val()
+                  ?.toString()
+                  .replace('T', ' ')
+                  .replace(/-/g, '/')}`,
```
Comment (konekoya) [inline ohara-manager/client/cypress/it/appBar/devTool.test.ts:307]: 🤯

File: ohara-manager/client/src/components/DevTool/Header/Controller/ControllerLog.js (hunk 4 above)
Comment (saivirtue) [reply ohara-manager/client/src/components/DevTool/Header/Controller/ControllerLog.js:218]: Sure thing :)

File: ohara-manager/client/cypress/it/appBar/devTool.test.ts (hunk 6 above)
Comment (saivirtue) [reply ohara-manager/client/cypress/it/appBar/devTool.test.ts:307]: The text format between `setting` and `statusBar` are different... 😞

Comment (konekoya) [reply ohara-manager/client/cypress/it/appBar/devTool.test.ts:307]: QQ

**Comment by saivirtue** [comment]
retry manager-it

**Comment by saivirtue** [comment]
retry manager-it

**Comment by saivirtue** [comment]
retry manager-it

**Comment by saivirtue** [comment]
retry manager-it

**Pull request merged**